kolena/_experimental/dataset/_dataset.py

```diff
@@ -36,6 +36,8 @@
 from kolena.workflow._datatypes import TypedDataObject
 from kolena.workflow.io import _dataframe_object_serde
 
+COLUMN_ORDER_FIELD = "_field_order"
```

Comment:
How will this field get populated? Seems we should add it to the base `DataObject._to_dict` implementation.

Reply:
With dataset's dataframe approach, only `datapoint` order info is needed. If individual columns contains annotation/asset (`TypedDataObject`), as they are deserialized back to each's own class, order is irrelevant. As a result, this field can be populated from the source dataframe.

Reply:
The Python client isn't the only place these objects are deserialized — having a deterministic column order in the app would be useful too. It's not a huge deal one way or the other, but since this approach can be applied to any data object, it seemed worth it.

![Screenshot 2023-10-16 at 10 22 55 AM](https://github.com/kolenaIO/kolena/assets/9648886/886b6170-40b9-4bf8-87dd-60ef9c05b89d)

Note that the frontend will need to be updated to handle this `_field_order` reserved key anyway.

Reply:
Are you thinking to add this also to annotation/asset for UI's purpose?

Reply:
Yes — anything that is serialized as a dict has its order lost at the persistence layer currently. If we're adding a mechanism to order top-level Kolena objects (like datapoint), it's a quick win to extend this pattern to nested Kolena objects (like assets and annotations).